input/pagecontent/reference.md

```diff
@@ -367,204 +447,238 @@ If the index is omitted, the first element of the array (`[0]`) is assumed.
 
   ```
   performer[Practitioner]
   ```
 
+* Path to the Reference(US Core Organization) option of the `performer` element in [US Core DiagnosticReport Lab](http://hl7.org/fhir/us/core/StructureDefinition-us-core-diagnosticreport-lab.html), using the canonical URL:
+
+  ```
+  performer[http://hl7.org/fhir/us/core/StructureDefinition/us-core-organization]
+  ```
+
+* The same path, using the id of the US Core Organization profile instead of its canonical URL:
+
+  ```
+  performer[us-core-organization]
+  ```
+
 #### Data Type Choice [x] Paths
 
-Addressing a type from a choice of types replaces the `[x]` in the property name with the type name (while also capitalizing the first letter). This follows the approach used in FHIR JSON and XML serialization.
+FHIR represents a choice of data types using `foo[x]` notation. To address a single data type, replace the `[x]` with the data type name (also capitalizing the first letter). To illustrate, Condition.onset[x], with choices dateTime, Age, Period, Range or string would have paths onsetDateTime, onsetAge, onsetPeriod, etc. This follows the approach used in FHIR JSON and XML serialization and is customary in FHIR.
+
+> **Note:** foo[x] choices are NOT addressed as foo[boolean], foo[Quantity], etc.
 
 **Example:**
 
-* Assign value[x] string value to "Hello World":
+* The path to the string data type of Observation.value[x]:
 
   ```
-  * valueString = "Hello World"
+  valueString
   ```
 
-* Assign the value[x] Reference value to an instance of the Patient resource:
+* The path to the Quantity within the SystolicBP component of the [Blood Pressure profile](https://www.hl7.org/fhir/bp.html):
 
   ```
-  * valueReference = Reference(EveAnyperson)
+  component[SystolicBP].valueQuantity
   ```
 
 #### Profiled Type Choice Paths
 
-In some cases, a type may be constrained to a set of possible profiles. To address a specific profile on that type, follow the path with square brackets (`[ ]`) containing the profile's `name`, `id`, or `url`.
+In some cases, a data type may be constrained to a set of possible profiles. To address a specific profile on that type, follow the path with square brackets (`[ ]`) containing the profile's `name`, `id`, or `url`.
 
 **Example:**
 
-* After constraining an address element to be either a USAddress or a CanadianAddress, bind the address.state properties to value sets of US states and Canadian provinces:
+* After constraining Patient.contact.address to be either a USAddress or a CanadianAddress (assuming these are profiles on Address), the paths to the respective `state` elements in the first contact:
 
   ```
-  * address only USAddress or CanadianAddress
-  * address[USAddress].state from https://www.usps.com
-  * address[CanadianAddress].state from http://ehealthontario.ca/fhir/CodeSystem/province-state-codes
+  contact[0].address[USAddress].state
+
+  contact[0].address[CanadianAddress].state
```

Comment:
There is only one Patient.contact.address element, true, but I believe you could bind a list of states to address[USAddress].state and a list of provinces to address[CanadianAddress].state. @cmoesel ?

Reply:
We haven't talked about this specific path syntax for a while.  @ngfreiter is right that it doesn't make sense in the context of profiling, because in the SD, it's still just a single element.  To apply additional constraints only for a specific profile, you would need to slice it by profile, and then use the slice path to apply the constraints.  You would _not_ use this profile path syntax for adding constraints.

But... in this example, the path already has `contact[0]`, which indicates to me that this is actually an `Instance` (_not_ a `Profile` definition).  In an `Instance`, this syntax _does_ make sense because it hints at which profile needs to be used for validation and filling in values.  For example, this rule is fine:
```
contact[0].address.state = USPS#CT
```
and this rule is also fine:
```
contact[0].address[USAddress].state = USPS#CT
```
but this rule is probably wrong (assuming `CanadianAddress` binds state to a value set that does not include `USPS#CT`):
```
contact[0].address[CanadianAddress].state = USPS#CT
```

And I just looked at [our old WIKI](https://github.com/HL7/fhir-shorthand/wiki/2.2-Paths#addressing-a-profile-from-a-profiled-type), and sure enough, this is exactly what we said -- that this path ONLY makes sense for `Instance`:

![image](https://user-images.githubusercontent.com/2278253/86640292-2d154100-bfa8-11ea-80f3-db5ab348ba95.png)

So... I think this part of the doc probably needs to be clarified, but it's not outright wrong (unless we want to drop this syntax entirely).

Reply:
I prefer to drop it entirely.